pkg/util/utils.go

```diff
@@ -723,3 +726,316 @@ func SocketPath() (string, error) {
 	// Glue the socket path together
 	return filepath.Join(xdg, "podman", "podman.sock"), nil
 }
+
+// ImageScpOptions provide options for securely copying images to and from a remote host
+type ImageScpOptions struct {
+	// Remote determines if this entity is operating on a remote machine
+	Remote bool `json:"remote,omitempty"`
+	// File is the input/output file for the save and load Operation
+	File string `json:"file,omitempty"`
+	// Quiet Determines if the save and load operation will be done quietly
+	Quiet bool `json:"quiet,omitempty"`
+	// Image is the image the user is providing to save and load
+	Image string `json:"image,omitempty"`
+	// Transfer determines whether we are transferring an image locally
+	Transfer bool `json:"transfer,omitempty"`
+	// User is used in conjunction with Transfer to determine if a valid user was given to save from/load into
+	User string `json:"user,omitempty"`
+}
+
+// ParseImageSCPArgs returns the valid connection data based off of the information provided by the user
+// args is an array of the command arguments and cfg is tooling configuration used to get service destinations
+// returned is serv and an error if applicable. serv is a map of service destinations with the connection name as the index
+// this connection name is intended to be used as EngineConfig.ServiceDestinations
+// this function modifies the global scpOpt entities: FromRemote, ToRemote, Connections, and SourceImageName
+func ParseImageSCPArgs(args []string) (*ImageScpOptions, *ImageScpOptions, []string, error) {
+	source := ImageScpOptions{}
+	dest := ImageScpOptions{}
+
+	cliConnections := []string{}
+	switch len(args) {
+	case 1:
+		if strings.Contains(args[0], "@localhost") { // image transfer between users
```

Comment:
Can you also do this as a switch statment.
Go can do 
switch {
case strings.Contains(args[0], "@localhost"):
...
case strings.Contains(args[0], "::"):
...
default:
...
}